test/server/config_validation/server_test.cc

```diff
@@ -11,23 +13,60 @@ namespace Server {
 // Test param is the path to the config file to validate.
 class ValidationServerTest : public testing::TestWithParam<std::string> {
 public:
-  static void SetUpTestCase() {
+  static void SetupTestDirectory() {
     TestEnvironment::exec(
         {TestEnvironment::runfilesPath("test/config_test/example_configs_test_setup.sh")});
     directory_ = TestEnvironment::temporaryDirectory() + "/test/config_test/";
   }
 
+  static void SetUpTestCase() { SetupTestDirectory(); }
+
 protected:
   ValidationServerTest() : options_(directory_ + GetParam()) {}
 
   static std::string directory_;
-
   testing::NiceMock<MockOptions> options_;
   TestComponentFactory component_factory_;
 };
 
 std::string ValidationServerTest::directory_ = "";
 
+// ValidationServerTest_1 is created only to run different set of parameterized
+// tests than set of tests for ValidationServerTest.
+class ValidationServerTest_1 : public ValidationServerTest {
+public:
+  static const std::vector<std::string> GetAllConfigFiles() {
+    SetupTestDirectory();
+
+    std::string all_file_names =
```

Comment:
Sorry for my edification, is it possible to use some of the common code in https://github.com/envoyproxy/envoy/blob/master/test/config_test/config_test.cc for this? Or not worth it?

Reply:
I will try to reuse common code.